src/ButtonToggle.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import Button from "./Button";
+import classNames from 'classnames';
+import { mapToCssModules } from './utils';
+
+const propTypes = {
+  onClick: PropTypes.func,
+  onBlur: PropTypes.func,
+  onFocus: PropTypes.func,
+  disabled: PropTypes.bool,
+}
+
+class ButtonToggle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      toggle: false,
+      focus: false,
+    }
+
+    this.onBlur = this.onBlur.bind(this);
+    this.onFocus = this.onFocus.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onBlur() {
+    this.props.onBlur();
+    this.setState({
+      focus: false,
+    });
+  }
+
+  onFocus() {
+    this.props.onFocus();
+    this.setState({
+      focus: true,
+    });
+  }
+
+  onClick(e) {
+    this.props.onClick();
```

Comment:
I think that we should allow the user's callback to be executed because it must be for some reason for which he/she must have written that code and passed down as `props` right?

Reply:
This is how it is handled elsewhere within the code:
https://github.com/reactstrap/reactstrap/blob/47843998b627aadcccfe71b092a194197fd2ce15/src/Button.js#L35-L44

I suppose it doesn't really matter since if you pass `disabled` to `Button`, then `Button` will not call your `onClick` anyways.